toptal/picasso
# fix(Page): fix hamburger menu not being visible on narrow viewports

[FX-3834](https://toptal-core.atlassian.net/browse/FX-3834)

### Description

A dropdown menu used to not be visible when `Page` was initially rendered in a narrow viewport.
Here's what happened:

- `PageHamburgerPortal` is responsible for getting DOM element to create a portal for the dropdown menu
- A callback located inside was called once for every component render, but before DOM updates are made, thus causing `getElementById` to return `null`
- This was patched by moving the call to the render method. Yes, it's ugly, but it's the best we can do without reinventing how hamburger menu works in Picasso

### How to test

- Compare Page stories on [master](https://picasso.toptal.net/?path=/story/layout-page--page) vs branch storybook (to be deployed). Make a viewport narrow enough to trigger the menu collapsing to the dropdown, then refresh the page
- CI should pass, no visual changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| ![image](https://user-images.githubusercontent.com/18409292/230895242-b90414ea-3b9c-441a-a154-25276368c257.png)| ![image](https://user-images.githubusercontent.com/18409292/230895318-001e20b6-e648-4d4d-bd6a-d77ba3f9d73f.png) |

### Development checks



> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
